src/matchers/general/toMatchSnapshot.ts

```diff
@@ -0,0 +1,28 @@
+import { toMatchSnapshot as jestToMatchSnapshot, toMatchInlineSnapshot as jestToMatchInlineSnapshot } from 'jest-snapshot'
+
+export async function toMatchSnapshot(received: any, propertiesOrHint?: object | string, hint?: string) {
```

Comment:
`received` should be from type `WebdriverIO.Element` and within the matcher we should get the `outerHTML` structure of the element. A nice feature would be to allow to ignore certain properties or all properties at all.

Reply:
I'm keeping the original jest.expect.toMatchSnapshot behavior here. For the element user can use like:

expect(element.outerHTML).toMatchSnapshot()

properly can implement expect(element).toMatchOuterHtmlSnapshot() later

Reply:
Totally agree with that after reviewing the other PR. Keeping the `toMatchSnapshot` as is makes sense to avoid surprises with users. I would call the new matcher `toMatchElementSnapshot`.